drivers/serial/uart_sam0.c

```diff
@@ -285,7 +297,7 @@ static const struct uart_sam0_dev_cfg uart_sam0_config_##n = {		       \
 	.baudrate = CONFIG_UART_SAM0_SERCOM##n##_CURRENT_SPEED,		       \
 	.pm_apbcmask = PM_APBCMASK_SERCOM##n,				       \
 	.gclk_clkctrl_id = GCLK_CLKCTRL_ID_SERCOM##n##_CORE,		       \
-	.ctrla = SERCOM_USART_CTRLA_RXPO(3) | SERCOM_USART_CTRLA_TXPO(1),      \
+	.ctrla = SERCOM_USART_CTRLA_RXPO(3) | 1 << SERCOM_USART_CTRLA_TXPO_Pos,\
```

Comment:
Use SERCOM_USART_CTRLA_TXPO_PAD2 instead.

Reply:
That one is only defined in the samd20 hal, not in samd21 :-(
I really like atmel's way of differentiating their SoC's.

Reply:
That one is only defined in the samd20 hal, not in samd21 :-(
I really like atmel's way of differentiating their SoC's.

Reply:
LGTM.